src/ex6/client/src/components/main-window.js

```diff
@@ -0,0 +1,34 @@
+import React, { useState, useEffect, useCallback } from "react";
+import { AddTodoForm } from "./add-todo-form.js";
+import { NoTodosPlaceholder } from "./no-todos-placeholder.js";
+import { TodosList } from "./todos-list.js";
+import { Footer } from "./footer.js";
+import { getItems } from "../item_client.js";
+
+export function MainWindow() {
+  const [todos, setTodos] = useState([]);
+  const [loaded, setLoaded] = useState(false)
+
+  const updateTodos = useCallback(async () => {
+    setTodos(await getItems());
+    setLoaded(true);
+  },[])
+
+  useEffect(() => {
+    updateTodos();
+  }, [updateTodos]);
+
+  return (
+    <main className="window">
+      <h1 className="title">TODOOPS</h1>
+      { loaded &&
```

Comment:
consider implementing a loading spinner to show the progress of the app + also give feedback to async operations like adding a new todo

Reply:
done